osquery/tables/system/darwin/signature.mm

```diff
@@ -23,6 +22,34 @@
 namespace osquery {
 namespace tables {
 
+// Get the flags to pass to SecStaticCodeCheckValidityWithErrors, depending on
+// the OS version.
+Status getVerifyFlags(SecCSFlags& flags) {
```

Comment:
If you keep a local static `SecCSFlags`, set it initially if it's non-0, you can skip selecting from the os_version every time. :)